### PR TITLE
`ThemeData` minor spring cleaning

### PR DESCRIPTION
**Pull request by guidezpl**
Do a bit of sprucing up of `ThemeData` in anticipation of future work

Related: https://github.com/flutter/flutter/issues/91772

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -1432,13 +1415,13 @@ class ThemeData with Diagnosticable {
...
-    // properties will gradually be phased out, see
-    // https://github.com/flutter/flutter/issues/91772.
+    ColorScheme? colorScheme,
     Brightness? brightness,
+    // [colorScheme] is the preferred way to configure colors. The [Color] properties
+    // listed below (as well as primarySwatch) will gradually be phased out, see
+    // https://github.com/flutter/flutter/issues/91772.
     Color? canvasColor,
     Color? cardColor,
-    ColorScheme? colorScheme,
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/theme_data.dart:1441]: The new ordering makes so much more sense 👍

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -136,23 +136,6 @@ abstract class ThemeExtension<T extends ThemeExtension<T>> {
...
-
-// The same video shows the splash compositing to #D7D7D7 on a background of
-// #E1E1E1. Again, assuming the splash has an opacity of 0x66, we can solve for
-// the actual color of the splash:
-const Color _kLightThemeSplashColor = Color(0x66C8C8C8);
-
-// Unfortunately, a similar video isn't available for the dark theme, which
-// means we assume the values in the spec are actually correct.
-const Color _kDarkThemeHighlightColor = Color(0x40CCCCCC);
-const Color _kDarkThemeSplashColor = Color(0x40CCCCCC);
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/theme_data.dart:154]: I think these comments could be very helpful for future reference. Could we possibly revert this change (or find another place to put this explanation)?

**Review by nate-thegrate — COMMENTED** [review]
I believe @TahaTesser is out this week, so hopefully it's all right for me to pop in.

I think it looks really good overall, just had one concern about documentation.

**Comment by Hixie** [comment]
test-exempt: code cleanup!

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/theme_data.dart:154]: I believe these were removed for future updates to ThemeData which would deprecate these colors, these shouldn't be anymore anyways.

**Review by TahaTesser — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/material/theme_data.dart:154]: Oh sweet—thanks for the review and the explanation! 🙂

**Comment by guidezpl** [comment]
thanks!

**Pull request merged**